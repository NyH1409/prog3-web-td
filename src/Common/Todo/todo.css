.main-container{
    margin: 5% 15%;
}

.todo-container{
    margin-top: 3%;
    padding: 3%;
}

.done-container{
    margin-top: 3%;
    padding: 3%;
}

.flex-container{
    display: flex;
    justify-content: space-between;
}

.to-check{
    margin-right: 10px;
}

.field{
    margin: auto;
    text-align: center;
}