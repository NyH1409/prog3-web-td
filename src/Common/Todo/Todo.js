import { Add } from "@mui/icons-material";
import { Button, Input } from "@mui/material";
import React, { useState } from "react";
import { Done } from "./components/Done";
import { Task } from "./components/Task";
import "./todo.css";


export function TodoList() {
    const [data, setData] = useState([]);
    const [done, setDone] = useState([]);
    const [inputValue, setInputValue] = useState("");

    function add() {
        const todo = {
            id : Math.random(),
            title : inputValue
        }
        if(todo?.title === "" || todo?.title === null){
            alert(`Task description required !!`)
        } else {
            setData((data)=>{
                return [...data,todo]
            })
        }
        setInputValue("")
    }

    function doneTask(task) {
        const remainTodo = data.filter((elt)=>{
            return elt.id !== task.id;
        })
        setData(remainTodo)
        setDone((done)=>{
            return [...done, task]
        })
    }
    return (
        <div className="main-container">
            <div className="field">
                <Input type="text" placeholder="Write the task decription here... "
                value={inputValue} 
                sx={{ width: "50%" }} 
                onChange={(e)=>setInputValue(e.target.value)}
                />
                <Button color="primary" variant="contained" 
                sx={{ textAlign: "center", marginLeft: 3 }} 
                startIcon={<Add/>} onClick={()=>add()}
                >Create</Button>
            </div>


            <div className="flex-container">
                <Task data ={data} doneTask={doneTask} title="In Progress"/>
                <Done done={done} title="Done"/>
            </div>
        </div>
    )
}