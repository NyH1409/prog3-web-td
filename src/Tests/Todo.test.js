import { fireEvent, render } from "@testing-library/react";
import  renderer  from "react-test-renderer";
import { TodoList } from "../Common/Todo/Todo";

describe("Test todo component", ()=>{
    it("Render correctly", ()=>{
        const component = renderer.create(
            <TodoList />
        )
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    })

    it("Create task ok", ()=>{
        const { getByTestId } = render(
            <TodoList />
        )
        const inputField = getByTestId("input-field").getElementsByTagName("input")[0];

        expect(inputField).toBeTruthy();
        expect(inputField.getAttribute("placeholder")).toEqual("Write the task decription here... ");
    })
})