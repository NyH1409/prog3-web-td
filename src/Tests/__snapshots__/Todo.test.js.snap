// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test todo component Render correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="field"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-oh84yh-MuiInputBase-root-MuiInput-root"
      data-testid="input-field"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Write the task decription here... "
        type="text"
        value=""
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1qgsd7h-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      Create
    </button>
  </div>
  <div
    className="flex-container"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 todo-container css-102lv6m-MuiPaper-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-c24xf0-MuiTypography-root"
      >
        TODO
      </h5>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 done-container css-8f5jog-MuiPaper-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-1bq8dcc-MuiTypography-root"
      >
        DONE
      </h5>
    </div>
  </div>
</div>
`;
